### PR TITLE
ddl: stabilize test TestIndexChange

### LINKED ISSUE #15531: index_change_test.go:testIndexChangeSuite.TestIndexChange failed
index_change_test.go:testIndexChangeSuite.TestIndexChange

Latest failed builds: 
https://internal.pingcap.net/idc-jenkins/job/tidb_ghpr_unit_test/28301/display/redirect

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

Issue Number: close #15531 

```
[2020-06-30T07:13:16.416Z] ----------------------------------------------------------------------

[2020-06-30T07:13:16.416Z] FAIL: index_change_test.go:54: testIndexChangeSuite.TestIndexChange

[2020-06-30T07:13:16.416Z] 

[2020-06-30T07:13:16.416Z] index_change_test.go:174:

[2020-06-30T07:13:16.416Z]     c.Check(errors.ErrorStack(checkErr), Equals, "")

[2020-06-30T07:13:16.416Z] ... obtained string = "" +

[2020-06-30T07:13:16.416Z] ...     "index should have been dropped\n" +

[2020-06-30T07:13:16.416Z] ...     "github.com/pingcap/tidb/ddl.(*testIndexChangeSuite).TestIndexChange.func2\n" +

[2020-06-30T07:13:16.416Z] ...     "\t/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/index_change_test.go:169\n" +

[2020-06-30T07:13:16.416Z] ...     "github.com/pingcap/tidb/ddl.(*TestDDLCallback).OnJobUpdated\n" +

[2020-06-30T07:13:16.416Z] ...     "\t/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/callback_test.go:72\n" +

[2020-06-30T07:13:16.416Z] ...     "github.com/pingcap/tidb/ddl.(*worker).handleDDLJobQueue\n" +

[2020-06-30T07:13:16.416Z] ...     "\t/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/ddl_worker.go:511\n" +

[2020-06-30T07:13:16.416Z] ...     "github.com/pingcap/tidb/ddl.(*worker).start\n" +

[2020-06-30T07:13:16.416Z] ...     "\t/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/ddl_worker.go:155\n" +

[2020-06-30T07:13:16.416Z] ...     "runtime.goexit\n" +

[2020-06-30T07:13:16.416Z] ...     "\t/usr/local/go/src/runtime/asm_amd64.s:1357"

[2020-06-30T07:13:16.416Z] ... expected string = ""

[2020-06-30T07:13:16.416Z] 

[2020-06-30T07:13:16.416Z] PASS: column_change_test.go:61: testColumnChangeSuite.TestColumnChange	0.971s
```
Problem Summary:

testCreateIndex may not finish. If we set `prevState = model.StateNone`, then in the first onJobUpdated, it may change prevState to StatePublic.

### What is changed and how it works?

Wait for the first onJobUpdated finish.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- No code

Side effects

### Release note 

- No release note

### CODE REVIEW

**Comment by wjhuang2016** [comment]
> Could you add the information that this test failed to PR description? I see there is only connections in the issue, but connections failed to open.

DONE

File: ddl/index_change_test.go [hunk 1]
```diff
@@ -124,10 +126,19 @@ func (s *testIndexChangeSuite) TestIndexChange(c *C) {
...
+			}
 		}
 	}
 	d.SetHook(tc)
 	testCreateIndex(c, ctx, d, s.dbInfo, originTable.Meta(), false, "c2", "c2")
+	for i := 0; i <= 10; i ++ {
+		if addIndexDone {
+			break
+		}
+		time.Sleep(50 * time.Millisecond)
```
Comment (bb7133) [inline ddl/index_change_test.go:140]: Don't get the point here, when `testCreateIndex()` is done, isn't `addIndexDone` has been set to `true` already?

Comment (wjhuang2016) [reply ddl/index_change_test.go:140]: No, The jobState may not be JobStateSynced

Comment (zimulala) [reply ddl/index_change_test.go:140]: Please add comments on why we need this check.

**Pull request merged**